### PR TITLE
feat[next]: enable DaCe test macro

### PR DESCRIPTION
**Pull request by edopao**
## Description

Enable all tests with DaCe backend and explicitly filter out those tests that are not supported yet.

## Requirements

- [ ] All fixes and/or new features come with corresponding tests.
- [ ] Important design decisions have been documented in the approriate ADR inside the [docs/development/ADRs/](docs/development/ADRs/Index.md) folder.

If this PR contains code authored by new contributors please make sure:

- [x] All the authors are covered by a valid contributor assignment agreement provided to ETH Zurich and signed by the employer if needed.
- [x] The PR contains an updated version of the `AUTHORS.md` file adding the names of all the new contributors.

### CODE REVIEW

File: tests/next_tests/integration_tests/feature_tests/iterator_tests/test_builtins.py [hunk 1]
```diff
@@ -331,6 +333,9 @@ def test_can_deref(program_processor, stencil):
 @pytest.mark.parametrize("as_column", [False, True])
 def test_cast(program_processor, as_column, input_value, dtype, expected_value):
     program_processor, validate = program_processor
+    if program_processor == run_dace_iterator:
+        pytest.xfail("Not supported in DaCe backend: cast")
```
Comment (edopao) [inline tests/next_tests/integration_tests/feature_tests/iterator_tests/test_builtins.py:337]: This test will fail in c++ compilation, I don't know if validation should catch this problem earlier.

File: tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_column_stencil.py [hunk 2]
```diff
@@ -318,11 +328,13 @@ def sum_fencil(out, inp0, inp1, k_size):
 
 def test_different_vertical_sizes_with_origin(program_processor):
     program_processor, validate = program_processor
+    if program_processor == run_dace_iterator:
+        pytest.xfail("Not supported in DaCe backend: origin")
```
Comment (edopao) [inline tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_column_stencil.py:332]: Same failure as above.

File: tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_column_stencil.py [hunk 3]
```diff
@@ -135,8 +145,10 @@ def ksum_fencil(i_size, k_start, k_end, inp, out):
 )
 def test_ksum_scan(program_processor, lift_mode, kstart, reference):
     program_processor, validate = program_processor
+    if program_processor == run_dace_iterator:
+        pytest.xfail("Not supported in DaCe backend: scan")
     shape = [1, 7]
-    inp = gtx.np_as_located_field(IDim, KDim)(np.asarray([list(range(7))]))
+    inp = gtx.np_as_located_field(IDim, KDim)(np.asarray([list(range(7))], dtype=np.float64))
```
Comment (edopao) [inline tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_column_stencil.py:151]: Adding type to be consistent with output and carry initial value.

File: tests/next_tests/integration_tests/feature_tests/iterator_tests/test_builtins.py [hunk 4]
```diff
@@ -331,6 +333,11 @@ def test_can_deref(program_processor, stencil):
 @pytest.mark.parametrize("as_column", [False, True])
 def test_cast(program_processor, as_column, input_value, dtype, expected_value):
     program_processor, validate = program_processor
+    if program_processor == run_dace_iterator:
+        # bug in DaCe translation to C-code with implicit cast on nested SDFG boundary
+        # this bug disappears after inline_sdfgs pass (part of simplify)
+        pytest.xfail("Not supported in DaCe backend: implicit cast")
```
Comment (edopao) [inline tests/next_tests/integration_tests/feature_tests/iterator_tests/test_builtins.py:339]: I propose to apply `inline_sdfgs` to the final SDFG, in order to enable several test cases. The `inline_sdfgs` pass is simple enough to be considered safe, and it should make compilation easier by reducing the level of nesting in function calls.

File: src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py [hunk 5]
```diff
@@ -534,7 +538,11 @@ def _visit_numeric_builtin(self, node: itir.FunCall) -> list[ValueExpr]:
         expr = fmt.format(*internals)
         node_type = self.node_types[id(node)]
         assert isinstance(node_type, itir_typing.Val)
-        type_ = itir_type_as_dace_type(node_type.dtype)
+        try:
+            type_ = itir_type_as_dace_type(node_type.dtype)
+        except NotImplementedError:
+            # return value of same type as operands, for example in max(a, b)
+            type_ = args[0].dtype
```
Comment (gronerl) [inline src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py:545]: This looks like it is just a work around for a problem in type inference?

File: src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py [hunk 6]
```diff
@@ -148,7 +148,11 @@ def builtin_if(
     expr = "({1} if {0} else {2})".format(*internals)
     node_type = transformer.node_types[id(node)]
     assert isinstance(node_type, itir_typing.Val)
-    type_ = itir_type_as_dace_type(node_type.dtype)
+    try:
+        type_ = itir_type_as_dace_type(node_type.dtype)
+    except NotImplementedError:
+        # the if expression is formatted with the return value as first argument
+        type_ = args[1].dtype
```
Comment (gronerl) [inline src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py:155]: Is this just a workaround for a bug in type inference?

File: src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py (hunk 5 above)
Comment (gronerl) [reply src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py:545]: @tehrengruber

File: tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_with_toy_connectivity.py [hunk 7]
```diff
@@ -205,8 +215,10 @@ def test_sparse_input_field(program_processor, lift_mode):
         assert np.allclose(out, ref)
 
 
-def test_sparse_input_field_v2v(program_processor, lift_mode):
-    program_processor, validate = program_processor
+def test_sparse_input_field_v2v(program_processor_no_dace_exec, lift_mode):
+    program_processor, validate = program_processor_no_dace_exec
+    if program_processor == run_dace_iterator:
+        pytest.xfail("Not supported in DaCe backend: assert fails")
```
Comment (gronerl) [inline tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_with_toy_connectivity.py:221]: Can we have a more meaningful error messasge here? Judging from the name, it is probably due to "sparse fields". (And other reasons possibly as well)

File: tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_with_toy_connectivity.py [hunk 8]
```diff
@@ -452,6 +464,8 @@ def test_sparse_shifted_stencil_reduce(program_processor_no_gtfn_exec, lift_mode
     if program_processor == gtfn.format_sourcecode:
         pytest.xfail("We cannot unroll a reduction on a sparse field only.")
         # With our current understanding, this iterator IR program is illegal, however we might want to fix it and therefore keep the test for now.
+    if program_processor == run_dace_iterator:
+        pytest.xfail("Not supported in DaCe backend: assert fails")
```
Comment (gronerl) [inline tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_with_toy_connectivity.py:468]:
```suggestion
        pytest.xfail("Illegal iterator IR")
```
I understand that you don't want to create a joint skip gtfn and dace fixture for this case, but maybe we can improve the error message.

**Review by gronerl — CHANGES_REQUESTED** [review]

File: tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_with_toy_connectivity.py (hunk 7 above)
Comment (edopao) [reply tests/next_tests/integration_tests/multi_feature_tests/iterator_tests/test_with_toy_connectivity.py]: In this case I can just rely on the dace fixture, and remove the additional error message.

File: src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py (hunk 5 above)
Comment (edopao) [reply src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py]: We agreed on disabling the failing tests. These issues in type inference will be addressed by #1271

File: src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py (hunk 6 above)
Comment (edopao) [reply src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py]: Same here.

**Review by gronerl — APPROVED** [review]
I have one comment about a error string, not imperative to fix though.

**Pull request merged**